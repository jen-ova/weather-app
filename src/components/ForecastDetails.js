import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import "../styles/weather-icons-wind.css";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, humidity, wind } = forecast;

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__temperature">
        {`Max temp: ${temperature.max}`}&deg;C
      </div>
      <div className="forecast-details__temperature">
        {`Min temp: ${temperature.min}`}&deg;C
      </div>
      <div className="forecast-details__humidity" data-testid="forecast-icon">
        Humidity: {humidity}%
      </div>
      <div className="forecast-details__wind">
        Wind: {wind.speed}mph
        <i className={`wi wi-wind wi-from-${wind.direction} wind-icon`} />
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
