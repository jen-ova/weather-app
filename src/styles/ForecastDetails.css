* {
    margin: 0.25rem;
}

.forecast-details {
    text-align: center;
    margin: 1rem;
}

.forecast-details__date {
    font-size: 2rem;
}

.wind-icon {
    font-size: 1rem;
    font-weight: bold;
}


