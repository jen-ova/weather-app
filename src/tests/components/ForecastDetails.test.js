import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    humidity: 30,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 10,
      direction: "s",
    },
  };

  it("renders", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails forecast={validProps} />
    );

    expect(getByText("Mon 30th Apr")).toHaveClass("forecast-details__date");
    expect(getByText("Max temp: 22°C")).toHaveClass(
      "forecast-details__temperature"
    );
    expect(getByText("Min temp: 12°C")).toHaveClass(
      "forecast-details__temperature"
    );
    expect(getByText("Humidity: 30%")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByText("Wind speed: 10mph")).toHaveClass(
      "forecast-details__wind"
    );
    expect(getByText("Wind direction: S")).toHaveClass(
      "forecast-details__direction"
    );
  });
});
