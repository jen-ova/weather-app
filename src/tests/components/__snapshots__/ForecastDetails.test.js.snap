// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon 30th Apr
    </div>
    <div
      class="forecast-details__temperature"
    >
      Max temp: 22°C
    </div>
    <div
      class="forecast-details__temperature"
    >
      Min temp: 12°C
    </div>
    <div
      class="forecast-details__humidity"
      data-testid="forecast-icon"
    >
      Humidity: 30%
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind speed: 10mph
    </div>
    <div
      class="forecast-details__direction"
    >
      Wind direction: S
    </div>
  </div>
</DocumentFragment>
`;
